document/js-api/index.bs

```diff
@@ -204,6 +204,27 @@ fetch('demo.wasm').then(response =>
 );
 ```
 
+<h2 id="webassembly-storage">Internal storage</h2>
+
+<h3 id="store">Interaction of the WebAssembly Store with JavaScript</h3>
+
+Note: WebAssembly semantics are defined in terms of a [=store=], representing the state of the world, or "memory". WebAssembly operations take a store and return a new store.
```

Comment:
Wording suggestion:

* Say "an abstract store" to clarify its role,
* Replace "state of the world" with "state of the WebAssembly abstract machine",
* Remove the "or 'memory'" to avoid confusion with Memory instances,
* Say "and return an updated store".